public/components/register_model/register_model.tsx

```diff
@@ -179,6 +182,22 @@ export const RegisterModelForm = () => {
     initializeForm();
   }, [latestVersionId, form]);
 
+  useEffect(() => {
+    if (!nameParams) {
+      return;
+    }
+    const subscriber = modelRepositoryManager
+      .getPreTrainedModel$(nameParams, 'torch_script')
+      .subscribe((preTrainedModel) => {
+        // TODO: store pre-trained model data
+        // eslint-disable-next-line no-console
+        console.log(preTrainedModel);
+      });
+    return () => {
+      subscriber.unsubscribe();
+    };
```

Comment:
Ideally, `useEffect` should return cleanup function, what about change it a bit like this?
```suggestion
    function subscribeToModel(name: string | null) {
      if (!name) {
        return;
      }
      return modelRepositoryManager
        .getPreTrainedModel$(name, 'torch_script')
        .subscribe((preTrainedModel) => {
          // TODO: store pre-trained model data
          // eslint-disable-next-line no-console
          console.log(preTrainedModel);
        });
    }
    const subscriber = subscribeToModel(nameParams);
    return () => {
      if (subscriber) {
        subscriber.unsubscribe();
      }
    };
```

Reply:
Are there any rules about this part? The code you suggested would be a bit complicated. I think the `useEffect` accept two forms of return value. If the name parameter exists, we will return the clean-up function which means will clean the subscribe when name changed. If it's not exists, we don't subscribe anything, we won't provide the clean-up function.